src/main/java/bio/terra/service/dataset/flight/delete/DeleteDatasetAzurePrimaryDataStep.java

```diff
@@ -58,15 +57,16 @@ public DeleteDatasetAzurePrimaryDataStep(
 
   @Override
   public StepResult doStep(FlightContext context) throws InterruptedException {
-    Dataset dataset = datasetService.retrieve(datasetId);
-    BillingProfileModel profileModel =
-        profileDao.getBillingProfileById(dataset.getDefaultProfileId());
-    AzureStorageAccountResource storageAccountResource =
-        resourceService.getDatasetStorageAccount(dataset, profileModel);
+    FlightMap map = context.getWorkingMap();
     AzureStorageAuthInfo storageAuthInfo =
-        AzureStorageAuthInfo.azureStorageAuthInfoBuilder(profileModel, storageAccountResource);
-    tableDao.deleteFilesFromDataset(
-        storageAuthInfo, f -> azureBlobStorePdao.deleteFile(f, userRequest));
+        map.get(DatasetWorkingMapKeys.AZURE_STORAGE_AUTH_INFO, AzureStorageAuthInfo.class);
+
+    if (storageAuthInfo != null) {
```

Comment:
Is this null check necessary? We have many other flights where one step depends on a previous step to set a value, but don't do an explicit null check, since there's no way the steps can run out of order. If the flight gets here and the previous step wasn't run, things are in a really bad state anyway.

If you do want to keep the null check, I would use `Objects.requireNonNull()`.
```
    AzureStorageAuthInfo storageAuthInfo =
        Objects.requireNonNull(map.get(DatasetWorkingMapKeys.AZURE_STORAGE_AUTH_INFO, AzureStorageAuthInfo.class), "No Azure storage auth info found");
```